tests/rustdoc/hidden-private.rs

```diff
@@ -0,0 +1,42 @@
+// This is a regression test for <https://github.com/rust-lang/rust/issues/106373>.
+// It ensures that the items in the `doc(hidden)` const block don't show up in the
+// generated docs.
+
+// compile-flags: --document-private-items
+
+#![crate_name = "foo"]
+
+// @has 'foo/index.html'
+// @count - '//*[@class="item-table"]//a[@class="struct"]' 2
+// @count - '//*[@class="item-table"]//a[@class="trait"]' 1
+#[doc(hidden)]
+const _: () = {
+    macro_rules! stry {
```

Comment:
Nit: Can you add a check that this doesn't appear in output

Reply:
Good idea.

Reply:
Done.